crates/bevy_reflect/bevy_reflect_derive/src/impls/structs.rs

```diff
@@ -99,16 +100,34 @@ pub(crate) fn impl_struct(reflect_struct: &ReflectStruct) -> TokenStream {
         bevy_reflect_path,
     );
 
-    let get_type_registration_impl = reflect_struct.get_type_registration();
+    let get_type_registration_impl = reflect_struct.get_type_registration(&field_types);
     let (impl_generics, ty_generics, where_clause) =
         reflect_struct.meta().generics().split_for_impl();
 
+    // Add Reflect bound for each active field
+    let mut where_reflect_clause = if where_clause.is_some() {
+        quote! {#where_clause}
+    } else if !field_types.is_empty() {
+        quote! {where}
+    } else {
+        quote! {}
+    };
+    where_reflect_clause.extend(quote! {
+        // TODO: had to add #bevy_reflect_path::Typed to get the test to compile,
+        //  presumably because of this:
+        //  #[inline]
+        //  fn get_type_info(&self) -> &'static bevy_reflect::TypeInfo {
+        //      <Self as bevy_reflect::Typed>::type_info()
+        //  }
```

Comment:
Is there no other way around this? We shouldn't need to specify this since the `Typed` impl guarantees `Self: Typed`. Or maybe I'm missing something 🤔

Reply:
I think it's because in the `Typed` derive, I had to add a `where T: Typed` clause.
It was not clear to me why I needed it.
Something in here:
```
            fn type_info() -> &'static bevy_reflect::TypeInfo {
                static CELL: bevy_reflect::utility::NonGenericTypeInfoCell = bevy_reflect::utility::NonGenericTypeInfoCell::new();
                CELL.get_or_set(|| {
                    let fields = [
                        bevy_reflect::NamedField::new::<u32>("a"),
                        bevy_reflect::NamedField::new::<String>("b"),
                    ];
                    let info = bevy_reflect::StructInfo::new::<Self>("Foo", &fields);
                    bevy_reflect::TypeInfo::Struct(info)
                })
            }
```
required it.

Reply:
For example, if I try with
```
#[derive(Reflect)]
struct Foo<T, U> {
    a: T,
    #[reflect(ignore)]
    b: U
}
```
 I get this error still:
```
error[E0277]: `U` cannot be shared between threads safely
164 |         struct Foo<T, U> {
    |                ^^^
note: required by a bound in `type_info::Typed`
   --> crates/bevy_reflect/src/type_info.rs:69:18
    |
69  | pub trait Typed: Reflect {
    |                  ^^^^^^^ required by this bound in `Typed`
```